ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -71,11 +72,19 @@ public synchronized BlockImportResult onBlock(final SignedBeaconBlock block) {
   }
 
   public AttestationProcessingResult onAttestation(
-      final MutableStore store, final Attestation attestation) {
+      final MutableStore store, final DelayableAttestation attestation) {
     return on_attestation(store, attestation, stateTransition, protoArrayForkChoiceStrategy);
   }
 
-  @Override
+  public void applyIndexedAttestation(
+          final MutableStore store, final DelayableAttestation attestation) {
+    protoArrayForkChoiceStrategy.onAttestation(store, attestation.getIndexedAttestation()
+            .orElseThrow(() -> new IllegalStateException("Future attestation should have had a proper indexed attestation."))
+    );
```

Comment:
I'd say this should just take an `IndexedAttestation` and push responsibility for ensuring `IndexedAttestation` all the way up to `AttestationManager`.  It shouldn't ever add an attestation to `FutureAttestation` that hasn't passed validation.

Reply:
In that case, would `FutureAttestation` only holding `IndexedAttestations` make sense?

Reply:
One downside is that maybe we want to pass the attestation back to the networking layer once we take it out from FutureAttestation. Also, keeping a reference to the `DelayableAttestation` (the higher-level object) is more in line with the `ValidateableAttestation` concept, so I might just keep items in FutureAttestation as DelayableAttestations.

Reply:
Done.